### PR TITLE
Enhance chatbot to handle case-insensitive keyword matching and impro…

### PR DESCRIPTION
**Pull request by Sisonke10-byt**
…ve response flexibility

- Modified the chatbot's response logic to support case-insensitive keyword matching.
- User input is now converted to lowercase to ensure it matches the keywords regardless of case.
- Keywords in the response objects are also converted to lowercase before comparison, allowing for flexible matching.
- This change ensures that the chatbot can handle user input in any case (e.g., "Hello", "hello", "HELLO") and still provide an appropriate response.
- Improved the overall flexibility of the chatbot by allowing it to respond to varied phrasing of the same query.
- The chatbot will now effectively understand a wider range of user inputs and return relevant responses.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -54,45 +54,89 @@ window.addEventListener('click', function(event) {
...
+    },
+    "linkedin": {
+        keywords: ["linkedin", "connect with me", "social media"],
+        response: "Connect with me on LinkedIn [here](https://www.linkedin.com/in/sisonkesamkele)."
+    },
+    "default": {
+        keywords: [],
+        response: "Sorry, I didn't understand that. Can you ask something else?"
+    }
 };
```
Comment (Andro-crt-prog) [inline script.js:101]: This is great work, your responses are more detailed.

File: script.js [hunk 2]
```diff
@@ -54,45 +54,89 @@ window.addEventListener('click', function(event) {
...
+        const { keywords, response } = chatbotResponses[category];
+        
+        // Check if the user's input contains any of the keywords
+        if (keywords.some(keyword => userInput.includes(keyword.toLowerCase()))) {
+            return response;
+        }
+    }
+    return chatbotResponses["default"].response;
+}
+
```
Comment (Andro-crt-prog) [inline script.js:134]: Nice work!

File: script.js (hunk 1 above)
Comment (Sisonke10-byt) [reply script.js:101]: Thank you @Andro-crt-prog

Comment (MaurenK) [reply script.js:101]: Looks professional, great work.

**Review by Andro-crt-prog — APPROVED** [review]

**Review by MolemoMM — APPROVED** [review]

**Comment by MaurenK** [comment]
Good code, very interesting to read.

**Pull request merged**